### PR TITLE
VP-1279 Add filtering to Contacts blade

### PR DESCRIPTION
**Pull request by AliveMen**
customer_filtering

### CODE REVIEW

File: VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                            <input type="checkbox" ng-model="blade.currentEntity.searchInVariations">
+                            <span class="switch"></span>
+                        </label>
+                    </div>
+                </div>
+            </form>
+            <div class="form-group">
+                <div class="list __info">
+                    <div class="list-t">org1.com</div>
+                    <div class="list-descr">all contacts with word 'org1.com' in property values</div>
```
Comment (yecli) [inline VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html]: Could you move these descriptions to localizations?

File: VirtoCommerce.CustomerModule.Web/Scripts/customer.js [hunk 2]
```diff
@@ -187,4 +212,12 @@ angular.module(moduleName, [])
               }]
           }
       });
+
+      // predefine search filters for customer search
+        predefinedSearchFilters.register(1583235535540, 'customerSearchFiltersDate', [
```
Comment (yecli) [inline VirtoCommerce.CustomerModule.Web/Scripts/customer.js]: What does this number mean - 1583235535540?

File: VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html (hunk 1 above)
Comment (yecli) [reply VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html]: Maybe need to tell what is 'property values' - e.g 'in searchable indexed property values (values in index '__content' array)'

File: VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                        <input ng-model="blade.currentEntity.name" required placeholder="{{ 'customer.blades.filter-detail.placeholders.name' | translate }}" />
+                    </div>
+                </div>
+                <div class="form-group">
+                    <label class="form-label">{{ 'customer.blades.filter-detail.labels.criteria' | translate }}</label>
+                    <div class="form-input">
+                        <input ng-model="blade.currentEntity.keyword" placeholder="{{ 'customer.blades.filter-detail.placeholders.criteria' | translate }}" />
+                    </div>
+                </div>
+                <div class="form-group">
```
Comment (yecli) [inline VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html]: Seems it is redundant for customers

File: VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                <div class="list __info">
+                    <div class="list-t">createddate:(2016-12-26 TO 2018-12-29)</div>
+                    <div class="list-descr">all contacts with creation period, between values excluding bounds</div>
+                </div>
+                <div class="list __info">
+                    <div class="list-t">modifieddate:[2016-12-26 TO]</div>
+                    <div class="list-descr">where the modified date is greater or equal than '2016-12-26'</div>
+                </div>
+                <div class="list __info">
+                    <div class="list-t">@org1.com membertype:Contact</div>
```
Comment (yecli) [inline VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html]: Empty result at the sample data (at least for me). Please use working example (without @org1.com)

File: VirtoCommerce.CustomerModule.Web/Scripts/customer.js (hunk 2 above)
Comment (AliveMen) [reply VirtoCommerce.CustomerModule.Web/Scripts/customer.js]: Last time modification date in milliseconds

File: VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html (hunk 1 above)
Comment (AliveMen) [reply VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html]: Moved

File: VirtoCommerce.CustomerModule.Web/Scripts/customer.js (hunk 2 above)
Comment (AliveMen) [reply VirtoCommerce.CustomerModule.Web/Scripts/customer.js]: The named constant created

File: VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html (hunk 3 above)
Comment (AliveMen) [reply VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html]: removed

File: VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html (hunk 4 above)
Comment (AliveMen) [reply VirtoCommerce.CustomerModule.Web/Scripts/blades/filter-detail.tpl.html]: fixed

**Comment by yecli** [comment]
Released: https://github.com/VirtoCommerce/vc-module-customer/releases/tag/v2.14.24

**Pull request merged**